pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1044,6 +1044,210 @@ spec:
+          type: string
+        - name: param2
+          type: string
+        steps:
+        - name: foo
+          image: busybox
+          script: 'exit 0'
+status:
+  pipelineSpec:
+    tasks:
+    - name: task1
+      taskSpec:
+        results:
+        - name: result1
+          type: string
+        - name: result2
+          type: string
+        steps:
+        - name: failing-step
+          onError: continue
+          image: busybox
+          script: 'echo -n 123 | tee $(results.result1.path); exit 1; echo -n 456 | tee $(results.result2.path)'
+    - name: task2
+      runAfter: [ task1 ]
+      params:
+      - name: param1
+        value: $(tasks.task1.results.result1)
+      - name: param2
+        value: $(tasks.task1.results.result2)
+      taskSpec:
+        params:
+        - name: param1
+          type: string
+        - name: param2
+          type: string
+        steps:
+        - name: foo
+          image: busybox
+          script: 'exit 0'
+  conditions:
+  - message: "Could not find result with name result2 for task task1"
+    reason: InvalidTaskResultReference
+    status: "False"
+    type: Succeeded
+  childReferences:
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: test-pipeline-missing-results-task1
+    pipelineTaskName: task1
+  provenance:
+    featureFlags:
+      RunningInEnvWithInjectedSidecars: true
+      EnableAPIFields: "stable"
+      AwaitSidecarReadiness: true
+      VerificationNoMatchPolicy: "ignore"
+      EnableProvenanceInStatus: true
+      ResultExtractionMethod: "termination-message"
+      MaxResultSize: 4096
+`)
+	d := test.Data{
+		PipelineRuns: prs,
+		TaskRuns:     trs,
+	}
+	prt := newPipelineRunTest(t, d)
+	defer prt.Cancel()
+
+	reconciledRun, clients := prt.reconcileRun("foo", "test-pipeline-missing-results", []string{}, true)
+	if reconciledRun.Status.CompletionTime == nil {
+		t.Errorf("Expected a CompletionTime on invalid PipelineRun but was nil")
+	}
+
+	// The PipelineRun should be marked as InvalidTaskResultReference.
+	if reconciledRun.Status.GetCondition(apis.ConditionSucceeded).Reason != "InvalidTaskResultReference" {
+		t.Errorf("Expected PipelineRun to be InvalidTaskResultReference but condition reason is %s", reconciledRun.Status.GetCondition(apis.ConditionSucceeded).Reason)
+	}
+	if d := cmp.Diff(reconciledRun, expectedPipelineRun, ignoreResourceVersion, ignoreLastTransitionTime, ignoreTypeMeta, ignoreStartTime, ignoreCompletionTime); d != "" {
+		t.Errorf("Expected to see PipelineRun run marked as failed. Diff %s", diff.PrintWantGot(d))
+	}
+
+	// Check that the expected TaskRun was created
+	taskrunList, err := clients.Pipeline.TektonV1beta1().TaskRuns("foo").List(prt.TestAssets.Ctx, metav1.ListOptions{
+		LabelSelector: "tekton.dev/pipelineRun=test-pipeline-missing-results",
+		Limit:         1,
+	})
+	if err != nil {
+		t.Fatalf("Error listing TaskRuns for PipelineRun %s: %s", prs[0].Name, err)
+	}
+	if len(taskrunList.Items) != 1 {
```

Comment:
can you use this helper instead? https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/pipelinerun/pipelinerun_test.go#L157-L166

Reply:
it would also be helpful to have a comment explaining that we don't want the pipelinerun to continue to create taskruns, which is why we're checking that there is still only 1 taskrun.

Reply:
Updated and added a comment :)